### PR TITLE
Automate publishing gem workflow

### PR DESCRIPTION
**Pull request by civitaspo**
* https://github.com/trocco-io/embulk-output-snowflake/commit/673486125b376f36a286a51238fe6972dfb357f4 Make sure the Git Tag matches the release version
* https://github.com/trocco-io/embulk-output-snowflake/commit/11de0585d8c8dbe17430013e9e8a42e7d01a057c Add a CI workflow to run lint,test,build,gemPush
* d367b3c Avoid "Invalid gemspec in [embulk-output-snowflake.gemspec]: Malformed version number string 11de058"

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      # TODO: Change this step after appling spotless.
+      # - name: lint
+      #   run: ./gradlew checkstyle
```
Comment (civitaspo) [inline .github/workflows/main.yml:26]: Enable this on the next PR that setup spotless.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Build Gem & Put it into './build/gems/' Directory
+        run: ./gradlew gem
+
+      - name: Set up JRuby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: jruby
+
+      - name: Publish to GPR
+        if: startsWith( github.ref, 'refs/tags/' )
```
Comment (civitaspo) [inline .github/workflows/main.yml:40]: This step runs only when tagging into this repository.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          touch $HOME/.gem/credentials
+          chmod 0600 $HOME/.gem/credentials
+          printf -- "---\n:github: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+          gem push --KEY github --host https://rubygems.pkg.github.com/${OWNER} ./build/gems/*.gem
+        env:
+          GEM_HOST_API_KEY: "Bearer ${{secrets.GITHUB_TOKEN}}"
+          OWNER: ${{ github.repository_owner }}
+
+      - name: Publish to RubyGems
+        if: startsWith( github.ref, 'refs/tags/' )
```
Comment (civitaspo) [inline .github/workflows/main.yml:52]: ditto

File: build.gradle [hunk 4]
```diff
@@ -3,13 +3,14 @@ plugins {
...
     id "org.embulk.embulk-plugins" version "0.4.2"
+    id "com.palantir.git-version" version "0.12.3"
 }
 repositories {
     mavenCentral()
 }
 
 group = "io.trocco"
-version = "0.3.1"
+version = versionDetails().lastTag
```
Comment (civitaspo) [inline build.gradle:13]: Rubygems does not allow pushing gems that have the same version as the past version, so gems cannot be released without tagging new one.

**Review by kekekenta — APPROVED** [review]
LGTM

File: .github/workflows/main.yml (hunk 2 above)
Comment (civitaspo) [reply .github/workflows/main.yml:40]: So, I have not been able to check the step works correctly. I will check when tagging the next.

**Comment by civitaspo** [comment]
@kekekenta thx

**Pull request merged**